### PR TITLE
fix- Can't disable "Show remaining" and "Show button time"

### LINKED ISSUE #14502: [BUG][2.17alpha][HEAD]: Can't disable "Show remaining" and "Show button time"
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

1. Use the HEAD ([0767aef](https://github.com/ankidroid/Anki-Android/commit/0767aef449e4dc7e0f83254a2f35b53f9929de15)) or recent one
2. "Settings">"Appearance"
3. (Confirm "Show top bar" is enabled. If it isn't, enable it.)
image

4. Tap "Show remaining"

### Expected behaviour

The "Show remaining" feature is switched to be disabled.

### Actual behaviour

The error message is shown. 
image

The "Show remaining" switch is visually toggled to be disabled, 
image
but the feature isn't actually disabled. 
image
On showing the "Appearance" screen again, you can see that the switch is toggled to be enabled.
image

https://github.com/ankidroid/Anki-Android/assets/10436072/5ae5c299-1e35-4846-b3a3-af08478f3560

### Debug info

```text
AnkiDroid Version = 2.17alpha1-debug (0767aef449e4dc7e0f83254a2f35b53f9929de15)

Android Version = 11

ProductFlavor = amazon

Manufacturer = OPPO

Model = CPH2013

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 11; CPH2013 Build/RKQ1.200903.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/116.0.0.0 Mobile Safari/537.36

ACRA UUID = f74d8c3b-b173-4337-ae0c-819d3934bd5e

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```

### (Optional) Anything else you want to share?

Not reproduced on both 2.16.5 and 2.17alpha1

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by vyom198**
## Purpose / Description
_Can't disable "Show remaining" and "Show button time" in the appearance section in settings_

## Fixes
Fixes #14502

## Approach
_Casting the value of newDueCountsValue and newETA to Boolean and passed the values to config.set() method_

## How Has This Been Tested?
_in the emulator_

https://github.com/ankidroid/Anki-Android/assets/112750331/d5441547-64a2-41c2-b78a-880a709842b8

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by lukstbit — APPROVED** [review]
LGTM, thank you for contributing!

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt [hunk 1]
```diff
@@ -134,7 +135,8 @@ class AppearanceSettingsFragment : SettingsFragment() {
         requirePreference<SwitchPreferenceCompat>(R.string.show_progress_preference).apply {
             launchCatchingTask { isChecked = withCol { config.get("dueCounts") ?: true } }
             setOnPreferenceChangeListener { newDueCountsValue ->
-                launchCatchingTask { withCol { config.set("dueCounts", newDueCountsValue) } }
+                val newDueCountsValueBool = newDueCountsValue as? Boolean ?: false
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt:138]: I suggest returning false inside the listener here, so the preference value isn't changed if the value for some reason is null.

Same for the above change

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt (hunk 1 above)
Comment (vyom198) [reply AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt:138]: should i make new pull request with the requested changes?

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt [hunk 2]
```diff
@@ -125,7 +125,8 @@ class AppearanceSettingsFragment : SettingsFragment() {
         requirePreference<SwitchPreferenceCompat>(R.string.show_estimates_preference).apply {
             launchCatchingTask { isChecked = withCol { config.get("estTimes") ?: true } }
             setOnPreferenceChangeListener { newETA ->
-                launchCatchingTask { withCol { config.set("estTimes", newETA) } }
+                val newETABool = newETA as? Boolean ?: false
+                launchCatchingTask { withCol { config.set("estTimes", newETABool) } }
             }
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt:130]: Here's what I mean with returning false at the listener
```suggestion
            setOnPreferenceChangeListener { _, newETA ->
                val newETABool = newETA as? Boolean ?: return@setOnPreferenceChangeListener false
                launchCatchingTask { withCol { config.set("estTimes", newETABool) } }
                true
            }
```

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt (hunk 1 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt:138]: See the comment below

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt (hunk 2 above)
Comment (vyom198) [reply AnkiDroid/src/main/java/com/ichi2/anki/preferences/AppearanceSettingsFragment.kt:130]: should i create new pull request with the requested changes?

**Review by BrayanDSO — APPROVED** [review]

**Pull request merged**